bundle/config/mutator/apply_presets.go

```diff
@@ -155,12 +155,25 @@ func (m *applyPresets) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnos
 
 	// Schemas: Prefix
 	for i := range r.Schemas {
-		prefix = "dev_" + b.Config.Workspace.CurrentUser.ShortName + "_"
-		r.Schemas[i].Name = prefix + r.Schemas[i].Name
+		schemaPrefix := "dev_" + b.Config.Workspace.CurrentUser.ShortName + "_"
+		r.Schemas[i].Name = schemaPrefix + r.Schemas[i].Name
 		// HTTP API for schemas doesn't yet support tags. It's only supported in
 		// the Databricks UI and via the SQL API.
 	}
 
+	// Clusters: Prefix, Tags
+	for _, c := range r.Clusters {
+		c.ClusterName = prefix + c.ClusterName
+		if c.CustomTags == nil {
+			c.CustomTags = make(map[string]string)
+		}
+		for _, tag := range tags {
+			if c.CustomTags[tag.Key] == "" {
+				c.CustomTags[tag.Key] = tag.Value
```

Comment:
These tags are seemingly freeform, but because they are passed through to the cluster VMs (thus the cloud provider), they are heavily constrained. We have a `Tagging` field on the bundle type that can be used for validation and/or normalization. I think it would be good to include this here such that users cannot pass invalid tags and then be surprised if the cluster doesn't start.